docs/writing-a-reconciler.md

```diff
@@ -0,0 +1,164 @@
+which provides the current state of the object, and the type of change that occurred (Create, Update, Delete). 
+For more information, see the [Asynchronous Business Logic section in Platform Concepts](application-design/platform-concepts.md#asynchronous-business-logic).
+
+## Reconciler vs Watcher
+
+Both reconcilers and watchers are used for the [reconciliation process](./application-design/platform-concepts.md#asynchronous-business-logic). 
+Whether you use one or the other is down to preference, and use-case. 
+Both reconcilers and watchers are powered by the same informer design within an `InformerController`, with just slightly different handling logic. 
+They both have an `Opinionated` variant that can wrap the interface as well.
+
+The major difference between a `Reconciler` and a `Watcher` is that a `Reconciler` has a single function, `Reconcile`, which is called for every event for a kind, while a `Watcher` has a function for each event type (`Add`, `Update`, `Delete`). There are more minor differences in how these events are handled as well:
+* A `Reconciler` can return a response with an explicit "retry after this time period" message, while a `Watcher` will only return success/fail (`nil` or `error`).
+* A `Watcher` will give you the previous state of the resource on an update event, while a Reconcile event will not.
+* A `Reconciler` can pass state between retries in-memory if storing state in the API server is failing, however, this will only persist until the operator is restarted and should not be relied upon excepting situations where the API server cannot be reached.
+
+## Considerations When Writing a Reconciler
+
+When writing a reconciler, it's important to take a few things into consideration:
+* If you make an update to the object you're doing the reconcile (or watch) event for, this will trigger _another_ reconcile (or watch) event. Generally, favor only updating subresources (specifically `status`) and some metadata in your reconcile (or watch) events, as a `status` update should not trigger the `metadata.generation` value to increase (only `metadata.resourceVersion`), which will allow you to filter events out. Using the `operator.OpinionatedWatcher` will filter these events for you, but you will need to track this yourself in a Reconciler; if you prefer not to use OpinionatedWatcher or want to do your own event filtering, keep in mind how updates within your reconcile loop will be received.
+* The reconciler is taking action on _every_ consumed event. Finding ways to escape from a reconcile or watcher event early will help your overall program logic.
+* All objects for the kind(s) you are watching are cached to memory by default. This can be customized by using a different informer implementation, such as `operator.CustomCacheInformer`. Custom informers can be used in `simple.App` with `AppConfig.InformerConfig.InformerSupplier`, or by using your own custom `app.App` implementation.
```

Comment:
`AppConfig.InformerConfig.InformerSupplier` is being added in [this PR](https://github.com/grafana/grafana-app-sdk/pull/612), and is not yet merged at time of writing.